### PR TITLE
Fix widget crashing on Android 12

### LINKED ISSUE #25: App crashes when trying to create a widget on Android 12
Log: https://hastebin.com/ugejusowoy.apache

### PR DESCRIPTION
**Pull request by FriederHannenheim**
Fixes #25

### CODE REVIEW

File: app/src/main/java/io/spaceapi/community/myhackerspace/Widget.java [hunk 1]
```diff
@@ -66,7 +69,7 @@ public void onDeleted(Context ctxt, int[] appWidgetIds) {
         for (int widgetId : appWidgetIds) {
             // Remove widget alarm
             PendingIntent pi = PendingIntent.getService(ctxt, widgetId,
-                    getIntent(ctxt, widgetId), 0);
+                        getIntent(ctxt, widgetId), SDK_INT >= Build.VERSION_CODES.S ? PendingIntent.FLAG_MUTABLE : 0);
```
Comment (niccokunzmann) [inline app/src/main/java/io/spaceapi/community/myhackerspace/Widget.java:72]: Cab you refactor the duplicated code into a method so we make sure future changes are also applied to all places?

Comment (FriederHannenheim) [reply app/src/main/java/io/spaceapi/community/myhackerspace/Widget.java]: Yep. Done

Comment (niccokunzmann) [reply app/src/main/java/io/spaceapi/community/myhackerspace/Widget.java]: Nice!

**Comment by dbrgn** [comment]
Does the widget actually work for you?

In the emulator with Android 12, if I add the widget to the homescreen, it's stuck on the loading animation:

screenshot-20220903-123027

I can dismiss the loading indicator, but the list of spaces remains empty.

(So far I did not debug the widget, and I don't actually know how it works, I mainly concentrated my efforts after forking into modernizing the main app.)

**Comment by dbrgn** [comment]
Ah, this seems to be #23. You're right, it just seems to be _very_ slow.

After adding the widget, it looks like this:

screenshot-20220903-123405

**Comment by niccokunzmann** [comment]
I think, since it is a small fix and I can not reproduce it, I would like to see it merged. It looks like it adds parameters only in case the condition of the OS version identified in the issue is used to change the default. So, I would trust the author with having good intentions and also invested the time.

**Comment by dbrgn** [comment]
You're right. From my point of view the widget was broken before and still doesn't work, but we can find out later what the reason for this is 🙂 Further fixes and improvements welcome.

**Comment by dbrgn** [comment]
I did a small cleanup commit for consistency (SDK_INT was imported statically, VERSION_CODES.S was not). Merging as soon as the build succeeds!

**Comment by FriederHannenheim** [comment]
> You're right. From my point of view the widget was broken before and still doesn't work, but we can find out later what the reason for this is 🙂 Further fixes and improvements welcome.

The widget is working. The loading just takes very long ~1 min but after that it works

**Pull request merged**